src/platform/datapath_raw_xdp_win.c

```diff
@@ -521,6 +427,7 @@ CxPlatDpRawInterfaceInitialize(
     for (uint8_t i = 0; i < Interface->QueueCount; i++) {
         XDP_QUEUE* Queue = &Interface->Queues[i];
 
+        Queue->RssProcessor = (uint16_t)Processors[i]; // TODO - Should memory be aligned with this?
```

Comment:
as in, should all allocations for this queue be aligned to that processor? ideally yes, for sure.

Reply:
Exactly, this is a larger issue too. For all memory allocation that is meant to be used on a particular process, should we temporarily affinitize the thread, allocate the memory and then revert the affinitization?

Reply:
in user mode, there are fancy APIs to allocate from a specific node, even if you're not currently running on that node. in kernel mode, it's more difficult to allocate pool from an arbitrary node, though i think newer OSes have better APIs. one sec.

that said, XDP itself treats the calling thread as the "ideal" thread when it performs its own allocations for queues, so temporarily setting the affinity would probably solve the problem the most universally and with the least fuss.

Reply:
yeah, PoolExtendedParameterNumaNode is supported for ExAllocatePool**3**. i'd still just set the thread and be done.

Reply:
So, if we're all on a single NUMA, how much would this actually matter?

Reply:
For VMs that are currently affixed to just one of the NUMA nodes, are we sure there's no per-proc memory perf involved? Also, can we be guaranteed that the VM actually knows the NUMA of the HW?

Reply:
AFAIK the VM can't do anything to optimize that scenario. It's all hidden and controlled by the host/hypervisor.

Reply:
Leaving it as a future TODO